crates/symbolicator/src/services/objects/data_cache.rs

```diff
@@ -71,7 +71,12 @@ impl ObjectHandle {
         match self.status {
             CacheStatus::Positive => Ok(Some(Object::parse(&self.data)?)),
             CacheStatus::Negative => Ok(None),
-            CacheStatus::Malformed => Err(ObjectError::Malformed),
+            CacheStatus::Malformed(MalformedCause::BadObject)
+            | CacheStatus::Malformed(MalformedCause::Unknown) => Err(ObjectError::Malformed),
+            CacheStatus::Malformed(MalformedCause::DownloadError) => {
+                // TODO: how can we restore the original error?
+                Err(ObjectError::Download(DownloadError::Canceled))
```

Comment:
We can't (or well, it might be possible but too complicated.  I'd try and add a message to the cache as explained above and then add a `DownloadError::Cached(String)` variant, that way we can propagate a details message.  But we don't need a full error message.  If it needed capturing into a sentry error it only needed capturing the non-cached version.

Reply:
resolved via the addition of the error string into the cache status